src/js/components/Meter/Bar.js

```diff
@@ -8,55 +8,169 @@ import { StyledMeter } from './StyledMeter';
 import { strokeProps, defaultColor } from './utils';
 
 const Bar = forwardRef((props, ref) => {
-  const { background, max, round, size, thickness, values, ...rest } = props;
+  const {
+    background,
+    max,
+    round,
+    size,
+    thickness,
+    direction,
+    values,
+    ...rest
+  } = props;
   const theme = useContext(ThemeContext) || defaultProps.theme;
-  const width =
-    size === 'full' ? 288 : parseMetricToNum(theme.global.size[size] || size);
-  const height = parseMetricToNum(
-    theme.global.edgeSize[thickness] || thickness,
-  );
+  let width;
+  let height;
+
+  if (direction === 'horizontal') {
+    width =
+      size === 'full' ? 288 : parseMetricToNum(theme.global.size[size] || size);
+    height = parseMetricToNum(theme.global.edgeSize[thickness] || thickness);
+  } else if (direction === 'vertical') {
+    height =
+      size === 'full' ? 288 : parseMetricToNum(theme.global.size[size] || size);
+    width = parseMetricToNum(theme.global.edgeSize[thickness] || thickness);
+  }
   // account for the round cap, if any
   const capOffset = round ? height / 2 : 0;
-  const mid = height / 2;
-  const someHighlight = (values || []).some(v => v.highlight);
+  let mid;
+  if (direction === 'horizontal') {
+    mid = height / 2;
+  } else if (direction === 'vertical') {
+    mid = width / 2;
+  }
 
+  const someHighlight = (values || []).some(v => v.highlight);
   let start = capOffset;
-  const paths = (values || [])
-    .filter(v => v.value > 0)
-    .map((valueArg, index) => {
-      const { color, highlight, label, onHover, value, ...pathRest } = valueArg;
-
-      const key = `p-${index}`;
-      const delta = (value * (width - 2 * capOffset)) / max;
-      const d = `M ${start},${mid} L ${start + delta},${mid}`;
-      const colorName =
-        color || defaultColor(index, theme, values ? values.length : 0);
-      let hoverProps;
-      if (onHover) {
-        hoverProps = {
-          onMouseOver: () => onHover(true),
-          onMouseLeave: () => onHover(false),
-        };
+
+  let paths;
+  if (direction === 'horizontal') {
```

Comment:
There's a fair amount of logic duplication with this coarse grained if/else approach. I'm wondering if we can merge these more, perhaps using a term like `length` for the calculations, and then just using the if/else on `direction` to construct the path.

Reply:
I am sorry, I didn't understand it. Can you please elaborate it bit further ?

Reply:
For example, in this set of changes, lines 63-89 are identical to lines 136-162. I think we should restructure the code to avoid that duplication. I am also wondering if we can do the calculations consistently and then only when constructing the actual path coordinates switch on horizontal vs. vertical. For example:

```
const d = (direction === 'horizontal')
  ? `M ${start},${mid} L ${start + delta},${mid}`
  : `M ${mid},${end - start} L ${mid},${(end - start) + delta}`
;
```
Note that this above code draws the vertical bar elements from the bottom, rather than handling that aspect in how the value of `start` is managed.

Reply:
I have refactored the codebase for the component and also added the prop in typescript `index.d.ts` file. Let me know if any other changes required.